### PR TITLE
Http4s blaze

### PR DESCRIPTION
**Pull request by richard-gibson**
_Before contributing, please read our [contributing guidelines](https://github.com/newrelic/newrelic-java-agent/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/newrelic/.github/blob/main/CODE_OF_CONDUCT.md)._

### Overview
This PR adds Instrumentation for blaze client and server implementations of the HTTP4S library. 

### Testing
Testing carried out in separate AITs

### Checks

[X] Are your contributions backwards compatible with relevant frameworks and APIs? Yes
[X] Does your code contain any breaking changes? No
[X] Does your code introduce any new dependencies? No

### CODE REVIEW

File: instrumentation/http4s-blaze-client-2.13_0.21/src/test/scala/com/nr/http4s/BlazeClientTest.scala [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    val introspector: Introspector = InstrumentationTestRunner.getIntrospector
+    val response = makeRequest(s"${server.getEndPoint}?no-transaction=1").unsafeRunTimed(2.seconds)
+
+
+    val traces = getTraces(introspector)
+    val segments = getSegments(traces)
+    val txnCount = introspector.getFinishedTransactionCount()
+    Assert.assertTrue("Web request successful", response.isDefined)
+//    Assert.assertEquals("Transaction count correct", 1, txnCount)
```
[the remark this answers is not in the document]
Comment (richard-gibson) [reply instrumentation/http4s-blaze-client-2.13_0.21/src/test/scala/com/nr/http4s/BlazeClientTest.scala:39]: thanks, removed

File: instrumentation/http4s-blaze-server-2.12_0.21/src/main/scala/com/nr/instrumentation/http4s/RequestWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  def getMethod: String = {
+    request.method.name
+  }
+
+  def getRequestURI: String = {
+    request.uri.path.toString()
+  }
+
+  def getRemoteUser: String = {
+    null
```
[the remark this answers is not in the document]
Comment (richard-gibson) [reply instrumentation/http4s-blaze-server-2.12_0.21/src/main/scala/com/nr/instrumentation/http4s/RequestWrapper.scala:23]: yes, not something I'd normally do. The `ExtendedRequest` class RequestWrapper implements is a Java class and there is no value to return for remoteUser. Looking at other usages for ExtendedRequest with getRemoteUser this seems to be the standard way of dealing with no value

**Review by twcrone — APPROVED** [review]

**Comment by twcrone** [comment]
Ran the new instrumentation tests and all functional tests successfully locally and did smoke test running against sample app provided by Richard.

**Comment by twcrone** [comment]
Also was able to run AITs against changes from this branch.

**Pull request merged**